src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -146,6 +192,24 @@ protected internal override async Task<HttpResponseMessage> SendAsync(HttpReques
 
                 requestObject.SetObjectProperty("method", request.Method.Method);
 
+                // Allowing or Disallowing redirects.
+                if (AllowAutoRedirect)
```

Comment:
Since this property is read on each request, I think the guards with `_handlerStarted` are unnecessary.

Reply:
This code was changed after other review discussion.  

`add flag to determine if the _allowAutoRedirect was explicitly set or not.` 

Does that change the evaluation?